### PR TITLE
fix(2488): fix pasted from WPS

### LINKED ISSUE #2488: Support for pasting content from WPS
I hope to copy and paste pictures and text in WPS software

### PR DESCRIPTION
**Pull request by yoyo837**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Base on https://github.com/ckeditor/ckeditor5/pull/10471, close #2488

---

### Additional information

I am a newcomer in here, please help to check.

@flairyu

### CODE REVIEW

**Comment by Reinmar** [comment]
I'm afraid that your test doesn't really test this scenario. The code coverage dropped.

Please use a more realistic markup – ideally an excerpt from a clipboard data that WPS generates.

**Comment by yoyo837** [comment]
> I'm afraid that your test doesn't really test this scenario. The code coverage dropped.
> 
> Please use a more realistic markup – ideally an excerpt from a clipboard data that WPS generates.

How should I get the clipboard content, if it is pasted directly to any editor, it loses markup.

**Comment by yoyo837** [comment]
> The code coverage dropped.

Yes, I am indeed trying to resolve CI Failed.

**Comment by Reinmar** [comment]
There are tools to inspect the clipboard's content. @Mgsy, what do you use normally?

**Comment by Mgsy** [comment]
I've created a simple snippet that allows viewing the clipboard HTML, feel free to use it - https://jsfiddle.net/m03co8fx/1/.

**Comment by yoyo837** [comment]
Example:

```html
<html xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:w="urn:schemas-microsoft-com:office:word" xmlns="http://www.w3.org/TR/REC-html40"><head><meta http-equiv=Content-Type content="text/html; charset=utf-8" ><meta name=ProgId content=Word.Document ><meta name=Generator content="Microsoft Word 14" ><meta name=Originator content="Microsoft Word 14" ><title></title><!--[if gte mso 9]><xml><w:WordDocument><w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel><w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery><w:DisplayVerticalDrawingGridEvery>2</w:DisplayVerticalDrawingGridEvery><w:DocumentKind>DocumentNotSpecified</w:DocumentKind><w:DrawingGridVerticalSpacing>7.8 磅</w:DrawingGridVerticalSpacing><w:PunctuationKerning></w:PunctuationKerning><w:View>Normal</w:View><w:Compatibility><w:DontGrowAutofit/><w:BalanceSingleByteDoubleByteWidth/><w:DoNotExpandShiftReturn/><w:UseFELayout/></w:Compatibility><w:Zoom>0</w:Zoom></w:WordDocument></xml><![endif]--><!--[if gte mso 9]><xml><w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="true" DefSemiHidden="true" DefQFormat="false" DefPriority="99" LatentStyleCount="260" >
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Normal" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 7" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 8" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="heading 9" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 7" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 8" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index 9" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 7" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 8" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toc 9" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Normal Indent" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="footnote text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="annotation text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="header" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="footer" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="index heading" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="caption" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="table of figures" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="envelope address" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="envelope return" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="footnote reference" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="annotation reference" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="line number" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="page number" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="endnote reference" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="endnote text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="table of authorities" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="macro" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="toa heading" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Bullet" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Number" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Bullet 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Bullet 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Bullet 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Bullet 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Number 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Number 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Number 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Number 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Title" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Closing" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Signature" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Default Paragraph Font" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text Indent" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Continue" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Continue 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Continue 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Continue 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="List Continue 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Message Header" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Subtitle" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Salutation" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Date" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text First Indent" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text First Indent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Note Heading" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text Indent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Body Text Indent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Block Text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Hyperlink" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="FollowedHyperlink" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Strong" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Emphasis" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Document Map" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Plain Text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="E-mail Signature" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Normal (Web)" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Acronym" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Address" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Cite" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Code" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Definition" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Keyboard" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Preformatted" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Sample" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Typewriter" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="HTML Variable" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Normal Table" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="annotation subject" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="No List" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Simple 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Simple 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Simple 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Classic 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Classic 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Classic 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Classic 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Colorful 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Colorful 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Colorful 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Columns 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Columns 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Columns 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Columns 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Columns 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 7" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid 8" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 7" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table List 8" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table 3D effects 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table 3D effects 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table 3D effects 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Contemporary" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Elegant" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Professional" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Subtle 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Subtle 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Web 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Web 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Web 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Balloon Text" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Grid" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Table Theme" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 1" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 2" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 3" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 4" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 5" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Shading Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light List Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Light Grid Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 1 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Shading 2 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 1 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium List 2 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 1 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 2 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Medium Grid 3 Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Dark List Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Shading Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful List Accent 6" ></w:LsdException>
<w:LsdException Locked="false" Priority="99" SemiHidden="false" Name="Colorful Grid Accent 6" ></w:LsdException>
</w:LatentStyles></xml><![endif]--><style>
@font-face{
font-family:"Times New Roman";
}

@font-face{
font-family:"宋体";
}

@font-face{
font-family:"等线";
}

p.MsoNormal{
mso-style-name:正文;
mso-style-parent:"";
margin:0pt;
margin-bottom:.0001pt;
mso-pagination:none;
text-align:justify;
text-justify:inter-ideograph;
font-family:等线;
mso-bidi-font-family:'Times New Roman';
font-size:10.5000pt;
mso-font-kerning:1.0000pt;
}

span.msoIns{
mso-style-type:export-only;
mso-style-name:"";
text-decoration:underline;
text-underline:single;
color:blue;
}

span.msoDel{
mso-style-type:export-only;
mso-style-name:"";
text-decoration:line-through;
color:red;
}
@page{mso-page-border-surround-header:no;
mso-page-border-surround-footer:no;}@page Section0{
}
div.Section0{page:Section0;}</style></head><body style="tab-interval:21pt;text-justify-trim:punctuation;" ><!--StartFragment--><p class=MsoNormal ><span style="mso-spacerun:'yes';font-family:等线;mso-bidi-font-family:'Times New Roman';font-size:10.5000pt;mso-font-kerning:1.0000pt;" ><font face="等线" >感受非遗魅力，弘扬传统文化，</font></span></p><!--EndFragment--></body></html>
```

Which part should I use? Or all?

File: packages/ckeditor5-paste-from-office/src/filters/space.js [hunk 1]
```diff
@@ -39,9 +39,13 @@ export function normalizeSpacing( htmlString ) {
  */
 export function normalizeSpacerunSpans( htmlDocument ) {
 	htmlDocument.querySelectorAll( 'span[style*=spacerun]' ).forEach( el => {
-		const innerTextLength = el.innerText.length || 0;
+		// For paste contents from WPS Office, the span's inner text is not all of spaces or blank,
+		// so we should keep the origin contents, or replace ONLY space characters.
+		if ( /[^\b]/.test( el.innerText.trim() ) === false ) {
```
Comment (Reinmar) [inline packages/ckeditor5-paste-from-office/src/filters/space.js:44]: What's the purpose of the `[^\b]` check here? What are we actually checking here? Couldn't it be replaced with `el.innerText.trim().length === 0`?

**Comment by Reinmar** [comment]
So the piece of HTML to be pushed through that function would be this:

```
<p class=MsoNormal ><span style="mso-spacerun:'yes';font-family:等线;mso-bidi-font-family:'Times New Roman';font-size:10.5000pt;mso-font-kerning:1.0000pt;" ><font face="等线" >感受非遗魅力，弘扬传统文化，</font></span></p>
```

I also left a question in a code comment because I don't understand the used check.

File: packages/ckeditor5-paste-from-office/src/filters/space.js (hunk 1 above)
Comment (yoyo837) [reply packages/ckeditor5-paste-from-office/src/filters/space.js:44]: Thanks for your reply. I will focus on this during the holidays. Please forgive my delay.

**Comment by yoyo837** [comment]
I need some help to fix the CI fields, please.

**Comment by yoyo837** [comment]
A more complete PR is here https://github.com/ckeditor/ckeditor5/pull/14039

**Pull request closed without merging**